zkevm-circuits/src/evm_circuit/execution/callop.rs

```diff
@@ -92,18 +85,26 @@ impl<F: Field> ExecutionGadget<F> for CallOpGadget<F> {
 
         let tx_id = cb.call_context(None, CallContextFieldTag::TxId);
         let mut reversion_info = cb.reversion_info_read(None);
-        let [current_address, is_static, depth] = [
-            CallContextFieldTag::CalleeAddress,
+        let [is_static, depth, current_address] = [
             CallContextFieldTag::IsStatic,
             CallContextFieldTag::Depth,
+            CallContextFieldTag::CalleeAddress,
         ]
         .map(|field_tag| cb.call_context(None, field_tag));
 
+        let [parent_address, parent_value] = cb.condition(is_delegatecall.expr(), |cb| {
+            [
+                CallContextFieldTag::CallerAddress,
+                CallContextFieldTag::Value,
+            ]
+            .map(|field_tag| cb.call_context(None, field_tag))
```

Comment:
We have to specify the parent call_id here. Otherwise, it would get the current address and value.

Reply:
The `caller_address` and `value` of current call is used to setup caller address and value for the `next` call (only for `DELEGATECALL`). I referenced [go-ethereum DELEGATECALL implmentation](https://github.com/ethereum/go-ethereum/blob/2b1299b1c006077c56ecbad32e79fc16febe3dd6/core/vm/evm.go#L323) and [function asDelegate](https://github.com/ethereum/go-ethereum/blob/2b1299b1c006077c56ecbad32e79fc16febe3dd6/core/vm/contract.go#L137).

I am sorry that these variable names seems not clear. I fixed to rename as:
```
current_address -> current_callee_address
parent_address -> current_caller_address
parent_value -> current_value
```